### PR TITLE
feat(template): add `toJSON`, `toArray` and `toObject` helpers

### PR DESCRIPTION
**Pull request by yueki-work**
## Changes
Add toJSON, toArray and toObject helpers to render complex json string easily

## Context
Closes #30046

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

https://github.com/yueki-work/renovate-pr-30633/pull/1

### CODE REVIEW

**Review by secustor — COMMENTED** [review]
Thanks for contributing, only doc changes necessary from my side.

File: docs/usage/templates.md [hunk 1]
```diff
@@ -39,6 +39,18 @@ If you want to print pretty JSON with Handlebars you can use the built-in functi
 
 In the example above `myvar` is a variable/field, that has valid JSON.
 
+### toJSON
```
[the remark this answers is not in the document]
Comment (yueki-work) [reply docs/usage/templates.md:42]: Thanks, fixed: 124ec4f39757df03f963a820a43a9602a3fbb57f

File: lib/util/template/index.spec.ts [hunk 2]
```diff
@@ -123,6 +123,12 @@ describe('util/template/index', () => {
     expect(JSON.parse(output)).toEqual({});
   });
 
+  it('to Object passing illegal number of elements', () => {
+    const userTemplate = "{{{ toJSON (toObject 'foo') }}}";
+    const outputFunc = () => template.compile(userTemplate, {});
+    expect(outputFunc).toThrow();
+  });
+
```
Comment (yueki-work) [inline lib/util/template/index.spec.ts:131]: Added test to increase coverage.
https://github.com/renovatebot/renovate/runs/28493167812

https://github.com/renovatebot/renovate/blob/48263f8db2b9e800f2de83fce11fce276992f81e/lib/util/template/index.ts#L35-L37

**Comment by yueki-work** [comment]
Sorry, checking lint failing

File: lib/util/template/index.ts [hunk 3]
```diff
@@ -16,6 +16,32 @@ handlebars.registerHelper('stringToPrettyJSON', (input: string): string =>
   JSON.stringify(JSON.parse(input), null, 2),
 );
 
+handlebars.registerHelper('toJSON', (input: any): string =>
```
Comment (viceice) [inline lib/util/template/index.ts:19]:
```suggestion
handlebars.registerHelper('toJSON', (input: unknown): string =>
```

File: lib/util/template/index.ts [hunk 4]
```diff
@@ -16,6 +16,32 @@ handlebars.registerHelper('stringToPrettyJSON', (input: string): string =>
...
+  JSON.stringify(input),
+);
+
+handlebars.registerHelper('toArray', (...args: any[]): any[] => {
```
Comment (viceice) [inline lib/util/template/index.ts:23]:
```suggestion
handlebars.registerHelper('toArray', (...args: unknown[]): unknown[] => {
```

File: lib/util/template/index.ts [hunk 5]
```diff
@@ -16,6 +16,32 @@ handlebars.registerHelper('stringToPrettyJSON', (input: string): string =>
...
+  // Need to remove the 'options', as last parameter
+  // https://handlebarsjs.com/api-reference/helpers.html
+  args.pop();
+  return args;
+});
+
+handlebars.registerHelper('toObject', (...args: any[]): any => {
```
Comment (viceice) [inline lib/util/template/index.ts:30]:
```suggestion
handlebars.registerHelper('toObject', (...args: unknown[]): unknown=> {
```

File: docs/usage/templates.md [hunk 6]
```diff
@@ -39,6 +39,24 @@ If you want to print pretty JSON with Handlebars you can use the built-in functi
...
+
+`{{{ toJSON upgrades }}}`
+
+### toObject
+
+If you want to convert key-value pairs to an object, use `toObject`, e.g.,
+
+`{{{ toJSON (toObject 'key1' 'value1' 'key2' 'value2') }}}` will render `{"key1":"value1","key2":"value2"}`.
+
+### toArray
```
Comment (viceice) [inline docs/usage/templates.md:54]: please move before `toJSON` for alphabetical order

**Review by viceice — APPROVED** [review]

**Comment by yueki-work** [comment]
>  the pr is tested with current main.

ah, I misunderstand this point, sorry.

**Pull request merged**